website/addons/owncloud/model.py

```diff
@@ -0,0 +1,153 @@
+# -*- coding: utf-8 -*-
+import logging
+from modularodm import fields
+from framework.auth import Auth
+from website.addons.base import exceptions
+from website.addons.base import StorageAddonBase
+from website.addons.base import (
+    AddonOAuthNodeSettingsBase, AddonOAuthUserSettingsBase,
+)
+from website.addons.owncloud.serializer import OwnCloudSerializer
+from website.addons.owncloud.settings import DEFAULT_HOSTS, USE_SSL
+from owncloud import Client as OwnCloudClient
+from website.addons.owncloud import settings
+
+from website.oauth.models import BasicAuthProviderMixin
+from website.util import api_v2_url
+
+logger = logging.getLogger(__name__)
+
+class OwnCloudProvider(BasicAuthProviderMixin):
+    """An alternative to `ExternalProvider` not tied to OAuth"""
+
+    name = 'ownCloud'
+    short_name = 'owncloud'
+    serializer = OwnCloudSerializer
```

Comment:
Alright, back on track with responses to the current round of updates. The provider certainly doesn't need a serializer, though I am still curious as to why the S3 provider needs one. In any case, in the current round, I'm experimenting with removing code blocks I added in to conform with the example plugins and this is one that it appears we can do without.